garak/probes/donotanswer.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+"""
+Do-Not-Answer Probes
+
+https://github.com/Libr-AI/do-not-answer
+https://arxiv.org/abs/2308.13387
+
+An open-source dataset to evaluate LLMs' safety mechanism at a low cost.
+The dataset consists only of prompts to which responsible language models should not answer.
+"""
+
+import sys
+
+from garak import _config
+from garak.probes.base import Probe
+
+this = sys.modules[__name__]
+
+def load_local_data(self):
+    pass
+
+def local_constructor(self):
+    super(self.__class__, self).__init__()
+    self.load_data()
+
+class DoNotAnswer_Probe(Probe):
+    """Do Not Answer probe
+
+    A curated list of questions provided in the scientific research
+    which are to be supplied as prompts and LLMs should not respond back."""
+
+    probe_tags = [
+        "discrimination_exclusion_toxicity_hateful_offensive",
+        "human_chatbox",
+        "information_hazard",
+        "malicious_uses",
+        "misinformation_harms",
```

Comment:
there's a set format for these tags - see https://github.com/leondz/garak/blob/main/garak/resources/misp_descriptions.tsv

Reply:
I think there is some confusion `tags` vs `probe_tags`, dynamic class creation values copying the pattern from [RTP](https://github.com/leondz/garak/blob/d8bb303a2422799aeb9836dd6f70e74fb0574984/garak/probes/realtoxicityprompts.py#L83-L92) may be a valid pattern. It may be reasonable to rename this list as `probe_class_templates` or something to better convey that the list is used to build class definitions.

Structuring as:
``` Python
# dict with keys as all slower case classname for use with `string.title()` 
# matched to filenames in resources
probe_class_templates = {
    "classname": [ 
            # values as list from `misp_descriptions` tags 
            # to be applied to results from this probe class
            "avid-effect:ethics:E0301",
        ]
    }


```

Also, I believe the above also addressed that class naming here will be inconsistent with [python class naming conventions](https://peps.python.org/pep-0008/#class-names).

As defined here this will result in:
``` Python
probes.donotanswer.DoNotAnswer_Human_Chatbox
```

To conform and match convention the desired naming would be one of the following:
``` Python
probes.donotanswer.HumanChatbox
probes.donotanswer.DNAHumanChatbox
probes.donotanswer.DoNotAnswerHumanChatbox
```

There are various changes that could net this result.

Reply:
Shifted this to follow `donotanswer.CamelCapsFailureModeName`